webdictionary/src/main/java/com/eis0/kademlia/KademliaId.java

```diff
@@ -146,7 +181,48 @@ public KademliaId xor(KademliaId nid) {
         return new KademliaId(result);
     }
 
+    /**
+     * Generates a NodeId that is some distance away from this NodeId
+     *
+     * @param distance The index of the first bit that should be different,
+     *                 From 0 to 159, where 0 changes the most significant bit
+     * @return The newly generated NodeId with a given distance from this.
+     * @throws IllegalArgumentException if the distance is < 0 or >= 160
+     * @author Edordo Raimondi, improved by Marco Cognolato
+     */
+    public KademliaId generateNodeIdByDistance(int distance) {
+        if(distance < 0 || distance >= ID_LENGTH) throw new IllegalArgumentException();
+
+        byte[] result = keyBytes.clone();
+
+        // calculate the index of the byte and bit to update
+        int byteToUpdateIndex = (distance) / 8;
+        int bitToUpdateIndex = 7 - (distance % 8);
```

Comment:
you're right, 8 is "8 bits"
The 7 is 8-1 but because when you count from 0 to 7 there are 8 values.
It's the same thing as "an array of length `n` has indeces from 0 to `n-1`"

I can definitely see why it can be considered hardwiring, though, is it right to put those in a const value?
I mean, a const value might be changed in the future by hand, but would it even make sense?
changing from 8 to for example 10 would mean that a byte is 10 bits, which clearly makes no sense.

That said, it might be better to put it in a const and clearly having a comment saying to not edit it, so I'll do it.